action.yml

```diff
@@ -23,38 +21,73 @@ runs:
-         formatted_labels=$(echo "$INPUT_LABELS" | sed -e 's/[][]//g' | awk -v RS=, -v ORS=, '{ gsub(/^[ \t\r\n]+|[ \t\r\n]+$/, "");  gsub(/\\"/, "\\\\\""); gsub(/"/, "\\\""); printf "\"%s\",", $0 }' | sed 's/,$//' | sed 's/^/[/; s/$/]/')
-         echo ' "labels": '"$formatted_labels">> /tmp/payload.json
+        # Set default operation to ADD if not provided
+        OPERATION=${INPUT_OPERATION:-ADD}
         
-        # operation argument is optional
-        if [ -n "$INPUT_OPERATION" ]; then
-          echo ', "operation": "'"$INPUT_OPERATION"'"' >> /tmp/payload.json
-        else
-          echo ', "operation": "'"REPLACE"'"' >> /tmp/payload.json
-        fi
+        # Perform API call based on operation
+        if [ "$OPERATION" = "ADD" ]; then
+          # Create payload (without operation)
+          echo '{ "artifact_id": "'"$INPUT_ARTIFACT_ID"'", ' > /tmp/payload.json
+          
+          # Take the value of the INPUT_LABELS variable, remove '[' and ']' characters,
+          # then process it to convert a comma-separated list of labels into a properly
+          # quoted JSON array, and store it in formatted_labels
+    
+          formatted_labels=$(
+            echo "$INPUT_LABELS" |                          # Output the value of INPUT_LABELS
+            sed -e 's/[][]//g' |                            # Remove all [ and ] characters
+            awk -v RS=, -v ORS=, '{                         # Split on commas (each label is a record)
+              gsub(/^[ \t\r\n]+|[ \t\r\n]+$/, "");          # Trim leading/trailing whitespace for each label
+              gsub(/\\"/, "\\\\\"");                        # Escape any existing backslash-quote \" as \\\"
+              gsub(/"/, "\\\"");                            # Escape regular quotes " as \"
+              printf "\"%s\",", $0                          # Output as: "label",
+            }' |
+            sed 's/,$//' |                                  # Remove the trailing comma after the last label
+            sed 's/^/[/; s/$/]/'                            # Add '[' at the start and ']' at the end, making a JSON array
+          )
+  
+          # Append the formatted labels as a JSON field to /tmp/payload.json
+          echo ' "labels": '"$formatted_labels"' }' >> /tmp/payload.json
+
+          # POST request to add labels
+          response=$(curl --fail-with-body -X 'POST' "$URL/v3/artifactinfos/$INPUT_ARTIFACT_ID/labels" \
+            -H "Authorization: Bearer $JWT" -H "Content-Type: application/json" \
+            --data-binary '@/tmp/payload.json') || command_failed=1
         
-        echo ' }' >> /tmp/payload.json
+        else # If operation is REMOVE
+
+          # Convert comma-separated labels to encoded query string
+          label_query=""
+          OLD_IFS="$IFS"
+          IFS=','
+          for label in $INPUT_LABELS; do
+            # Remove leading/trailing whitespace and encode spaces as %20:
+            label_trimmed=$(echo "$label" | xargs)
```

Comment:
same for delete operation with characters such as leading and trailing blank spaces and commas. With the current logic, we cannot delete a label which happens to have a leading blank space. We can add such labels: https://saas-preprod.beescloud.com/cloudbees-preprod/f56b861b-e24b-411d-4a35-c3c2c1c277f0/components/603c81e4-6571-49c8-bd59-65a6ba4b0ea7/artifacts?fullHistory=false&organizationId=202377bf-a896-419c-638f-c1ecb66be270.